core/src/main/java/com/linecorp/armeria/client/ClientTlsConfig.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.AbstractTlsConfig;
+import com.linecorp.armeria.common.TlsProvider;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.metric.MeterIdPrefix;
+
+import io.netty.handler.ssl.SslContextBuilder;
+
+/**
+ * Provides client-side TLS configuration for {@link TlsProvider}.
+ */
+@UnstableApi
+public final class ClientTlsConfig extends AbstractTlsConfig {
```

Comment:
Did you imagine to return the simplified `TrustManager` instead of `List<X509Certificate>` for `trustedCertificates()`?

I hesitated to add the simplified `TrustManager` because `TlsProvider` was designed to use both server and client but `tlsNoVerifySet` and `insecureHosts` would be only useful for clients. Moreover, they are not a recommended option for production.

Reply:
> Did you imagine to return the simplified TrustManager instead of List<X509Certificate> for trustedCertificates()?

Right, I imagined that `TlsProvider` contains a `trustManager()`-esque API

> I hesitated to add the simplified TrustManager because TlsProvider was designed to use both server and client but tlsNoVerifySet and insecureHosts would be only useful for clients. Moreover, they are not a recommended option for production.

I thought `TrustManager` is used by both server and client side. 
I still find it odd that `TrustManager` related APIs exist in both APIs (`TlsProvider`, `ClientTlsConfig`), but I'd rather not delay this PR

Reply:
I think TlsProvider's function is to provide the data required for TLS handshake rather than associating with `TrustedManger`. 

`tlsNoVerifySet`, `insecureHosts` are options that exist at the code level so it would be ambiguous to express them as data. I thought `ClientTlsConfig` was a better place to set them. 

I understand your concerns and this API may cause confusion or inconvenience. Let's get user feedback and decide whether the current design needs refactoring.